server/src/bin/dump-schema.rs

```diff
@@ -0,0 +1,37 @@
+use anyhow::Result;
+use juniper::{introspect, IntrospectionFormat};
+use stacks_exchange::{
+    db::make_pool,
+    graphql::{make_schema, Context},
+    settings::Settings,
+};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "dump_schema")]
+struct Opt {
+    #[structopt(parse(from_os_str), short, long)]
+    conf: Option<PathBuf>,
+}
+
+fn main() -> Result<()> {
+    dotenv::dotenv().ok();
+    let opt = Opt::from_args();
+    let settings = Settings::init(opt.conf)?;
+    pretty_env_logger::init();
+
+    let pool = make_pool(&settings.db)?;
+
+    let (res, _errors) = introspect(
+        &make_schema(),
+        &Context::new(settings, pool),
+        IntrospectionFormat::default(),
+    )
+    .unwrap();
+
+    let json = serde_json::to_string_pretty(&res)?;
```

Comment:
Not sure about nanoserde, but I trust the author of miniserde.

Reply:
I might be biased because he works for Facebook and I've met him, but he's the author of [many of the most popular crates in the Rust ecosystem](https://crates.io/users/dtolnay?sort=downloads), including anyhow and serde itself

Reply:
The conversion of config::Config to Settings depends on serde::Deserialize :-\ The good news is that build times aren't too awful. Clean builds take a while, but rebuilds usually finish in around 10 seconds